worldedit-core/src/main/java/com/sk89q/worldedit/world/block/BlockType.java

```diff
@@ -248,18 +248,17 @@ public boolean hasProperty(PropertyKey key) {
     }
 
     /**
-     * {@return whether this block type has a property with given key of the given type}
+     * {@return whether this block type has a given property}
      *
-     * @param key          the key identifying the property
-     * @param propertyType the expected type of the property
+     * @param property     the expected property
      * @since TODO
      */
-    public boolean hasPropertyOfType(PropertyKey key, Class<?> propertyType) {
-        int ordinal = key.getId();
-        Property<?> property;
+    public boolean hasProperty(Property<?> property) {
+        int ordinal = property.getKey().getId();
+        Property<?> selfProperty;
         return this.settings.propertiesMapArr.length > ordinal
-                && (property = this.settings.propertiesMapArr[ordinal]) != null
-                && property.getClass() == propertyType;
+                && (selfProperty = this.settings.propertiesMapArr[ordinal]) != null
+                && selfProperty == property;
```

Comment:
I don't think this is sufficient. For some properties, we create them on the spot (i.e. EnumProperty and DirectionalProperty), so different instances might represent the same actual property. I think using `equals` and overriding it in the relevant property classes is more reliable.

The null check can be removed too I guess, then the assignment isn't needed anymore either.

Reply:
Yep, I improved the equals in AbstractProperty and then forgot to use it anyway :)